File: main/Python/src/pkg/bluetooth_server.py
class BluetoothServer:

# public:

    def __init__(self, ser_):
        self.__m_message = ""

        self.__m_id = 0
        self.__m_max_id = 1000

        self.__m_airPressure = []

        self.__m_ser = ser_

    def setAirPressureRslt(self, rslt_):
        self.__m_airPressure = rslt_

    def responseMessage(self):

        self.__m_message = f"response predictedAirPressure value {' '.join(map(str, self.__m_airPressure[0]))}"
        self.__incrementId()
        self.__appendId()
        self.__appendDelimiter("\n")

        self.__m_ser.write(self.__m_message.encode("utf-8"))
        print("Python send :" + self.__m_message)

    def resendMessage(self):

        self.__m_message = f"response predictedAirPressure value {' '.join(map(str, self.__m_airPressure))}"
        self.__appendId()
        self.__appendDelimiter("\n")

        self.__m_ser.write(self.__m_message.encode("utf-8"))
        print("Python send :" + self.__m_message)

    def getId(self):
        return self.__m_id

# private:
    
    def __incrementId(self):
        self.__m_id = (self.__m_id % self.__m_max_id) + 1
    
    def __appendId(self):
        self.__m_message = f"{self.__m_message} id {self.__m_id}"
    
    def __appendDelimiter(self, delimiter="\n"):
        self.__m_message = self.__m_message + delimiter

File: main/Arduino/datacollect/bluetooth_server.cpp
#include "bluetooth_server.hpp"

namespace Dcon
{

namespace Server
{

BluetoothServer::BluetoothServer()
{

}

void BluetoothServer::init()
{
  m_message = " ";

  m_id = 0;
  m_max_id = 0;

  m_loadcell[LOAD_CELL_ARRAY_SIZE] = {0};

}

void BluetoothServer::setLoadCellRslt(const float rslt_[])
{
  for (int i = 0; i < LOAD_CELL_ARRAY_SIZE; i++)
  {
    m_loadcell[i] = rslt_[i];
  }
}

void BluetoothServer::responseMessage()
{
  m_message = "response loadcell value";

  for (int i = 0; i < LOAD_CELL_ARRAY_SIZE; i++)
  {
    m_message = m_message + " " + String(m_loadcell[i]);
  }

  incrementId();
  appendId();
  appendDelimiter("\n");
  
  Serial.print(m_message);
}

void BluetoothServer::resendMessage()
{
  m_message = "response loadcell value";

  for (int i = 0; i < LOAD_CELL_ARRAY_SIZE; i++)
  {
    m_message = m_message + " " + String(m_loadcell[i]);
  }

  appendId();
  appendDelimiter("\n");
  
  Serial.print(m_message);
}

int16_t BluetoothServer::getId()
{
  return m_id;
}

void BluetoothServer::incrementId()
{
  m_id = (m_id % m_max_id) + 1;
}

void BluetoothServer::appendId()
{
  m_message = m_message + " id " + String(m_id); 
}

void BluetoothServer::appendDelimiter(const char* delimiter)
{
  m_message = m_message + *delimiter;
}

}

}